integration_test/src/test_runner/mod.rs

```diff
@@ -302,13 +310,46 @@ fn run_function_test(
             unsafe {
                 Ok(odbc_sys::SQLTablesW(
                     statement.handle() as HStmt,
-                    wstr_or_null(&function[1]),
+                    wstr_or_null(&function[1]).0,
+                    to_i16(&function[2])?,
+                    wstr_or_null(&function[3]).0,
+                    to_i16(&function[4])?,
+                    wstr_or_null(&function[5]).0,
+                    to_i16(&function[6])?,
+                    wstr_or_null(&function[7]).0,
+                    to_i16(&function[8])?,
+                ))
+            }
+        }
+        "sqlcolumns" => {
+            check_array_length(function, 9)?;
+            unsafe {
+                Ok(odbc_sys::SQLColumns(
+                    statement.handle() as HStmt,
+                    str_or_null(&function[1]),
```

Comment:
the `SQLText::ptr` function used by `str_or_null` uses `mem::transmute`... so it does not have the same issue. It may or may not be leaking memory. I have to investigate further to be sure.